### PR TITLE
fix: restore adapter state on checkpoint resume

### PR DESCRIPTION
**Pull request by tusharui**
## Problem

When using `ModelCheckpoint` to resume a search, the crossover/mutation adapter step counters (`current_step`, `current_value`) were not saved or restored. This caused adaptive probability schedules (`ExponentialAdapter`, `InverseAdapter`, `PotentialAdapter`) to restart from step 0 instead of continuing from where the previous run left off.

## Reproducible Example

from sklearn.datasets import load_iris
from sklearn.tree import DecisionTreeClassifier
from sklearn_genetic import GASearchCV
from sklearn_genetic.space import Integer
from sklearn_genetic.callbacks.model_checkpoint import ModelCheckpoint
from sklearn_genetic.schedules.schedulers import ExponentialAdapter

X, y = load_iris(return_X_y=True)

def build():
    return GASearchCV(
        estimator=DecisionTreeClassifier(random_state=0),
        param_grid={"max_depth": Integer(1, 3)},
        cv=2, scoring="accuracy", population_size=4, generations=3,
        crossover_probability=ExponentialAdapter(0.8, 0.2, 0.1),
        mutation_probability=ExponentialAdapter(0.6, 0.1, 0.1),
    )

first = build()
first.fit(X, y, callbacks=ModelCheckpoint(checkpoint_path="cp.pkl"))
print(first.crossover_adapter.current_step)  # 3

resumed = build()
resumed.fit(X, y, callbacks=ModelCheckpoint(checkpoint_path="cp.pkl"))
print(resumed.crossover_adapter.current_step)
# Expected: 6  (3 + 3 more generations)
# Before fix: 3  (restarted from 0)
## Root Cause
ModelCheckpoint.on_step() saves _checkpoint_state() (constructor params) and runtime_state (fitness cache, logbook, fit stats) — but not the adapter objects. On resume, __dict__.update() restores constructor params while _reset_adapters() is skipped (checkpoint_loaded=True), leaving adapters stuck at current_step=0.
Fix
- schedules/base.py — Add state_dict() / load_state_dict() to BaseAdapter for serializing mutable state.
- callbacks/model_checkpoint.py — Save adapter state into runtime_state["adapter_state"].
- genetic_search.py — Restore adapter state on resume in both GASearchCV.fit() and GAFeatureSelectionCV.fit().
- Tests — 3 new tests covering adapter checkpoint resume for both estimators and state_dict round-trip.
Backward Compatibility
Old checkpoints without adapter_state in runtime_state still work — adapters default to step 0 (same behavior as before this fix).
Screenshot (400)

### CODE REVIEW

**Comment by rodrigo-arenas** [comment]
Thanks for the contribution and the clear write-up, @tusharui! Reviewed the fix and it correctly addresses the root cause — adapters are restored via `load_state_dict()` onto the existing instances after resume, with sensible backward compatibility for old checkpoints without `adapter_state`. Verified locally: all tests pass, including the new adapter-resume coverage for both `GASearchCV` and `GAFeatureSelectionCV`. Looks good to merge.
